demo/api/src/main.ts

```diff
@@ -43,6 +43,11 @@ async function bootstrap(): Promise<void> {
     app.use(compression());
     app.use(cookieParser());
 
+    // if CDN is enabled, make sure all traffic is either coming from the CDN or internal sources
+    if (config.cdn.enabled) {
```

Comment:
could you replace that with `config.cdn.originCheck`? And remove cnd.enabled?

Reply:
I've done this on purpose to enable easier toggling of this feature.

Reply:
imho less config options and env vars should be preferred if possible

Reply:
I think a more explicit approach is better for easier understanding

@johnnyomair what do you think?

Reply:
as discussed, we remove enabled